bottom/__init__.py

```diff
@@ -11,42 +13,66 @@
 class Client(event.EventsMixin):
     __conn_cls = connection.Connection
 
-    def __init__(self, host, port, *, encoding='UTF-8', ssl=True, loop=None):
+    def __init__(self, host: str, port: str, *, encoding: str = 'UTF-8',
+                 ssl: bool = True, loop: asyncio.BaseEventLoop = None) -> None:
+        """
+        :param host: The IRC server address to connect to.
+        :param port: The port of the IRC server.
+        :param encoding: The character encoding to use.
+        :param ssl: Whether SSL should be used while connecting.
+        :param loop: The event loop to use. Omit unless you know what you're
+        doing.
+        """
         if loop is None:
             loop = asyncio.get_event_loop()
         super().__init__(unpack.parameters, loop=loop)
         self.connection = self.__conn_cls(host, port, self, ssl=ssl,
                                           encoding=encoding, loop=loop)
 
-    def send(self, command, **kwargs):
+    def send(self, command: str, **kwargs) -> None:
         """
         Send a message to the server.
 
-        Examples
-        --------
-        client.send('nick', nick='weatherbot')
-        client.send('privmsg', target='#python', message="Hello, World!")
+        Examples:
+
+        .. code-block:: python
+
+            client.send('nick', nick='weatherbot')
+            client.send('privmsg', target='#python', message="Hello, World!")
 
         """
         packed_command = pack.pack_command(command, **kwargs)
         self.connection.send(packed_command)
 
-    async def connect(self):
+    async def connect(self) -> None:
```

Comment:
I find this interesting, since async functions don't return anything in a usable sense*.  Is this adding information, or just for conformity that an IDE plugin (or other consumer) expects?  Couldn't consumers of type information just assume a return of None?

* They return a coroutine object, from a coroutine function, but we almost never hang onto the returned objects from the call